Automattic/remote-data-blocks
# Improves Settings UI

- Refactored the current settings page to have a consistent styling and avoid page layout shifts.
- Improves validation, success and error messages.
- Refactored stylesheets i.e. each component have a dedicated file now

### How to test?

- Go through all possible screens of plugin and make sure everything is working as expected.
- If you face PHP error of converting stdClass to array then run `git revert a1b6bc8144060659f8b0dc448701566004842b77` to resolve the issue.


### Screenshots

<img width="2053" alt="1  Loader" src="https://github.com/user-attachments/assets/096ca7cd-83ea-4f08-b138-24a6885c269c">

<img width="2054" alt="2  Empty State" src="https://github.com/user-attachments/assets/bfa429e9-45ec-41b7-87d4-b07a1651c7cc">

<img width="2054" alt="3  List View" src="https://github.com/user-attachments/assets/840fc010-a4bd-44fe-b2bd-667c16c3e864">

<img width="2056" alt="4  Add Data Source" src="https://github.com/user-attachments/assets/8e5293f5-3654-4d23-bf55-78af266906d4">

<img width="2054" alt="5  Delete Data Source" src="https://github.com/user-attachments/assets/2b73bee3-30d7-4131-9111-028ff0d71f57">

<img width="2051" alt="6  Add Airtable Data Source" src="https://github.com/user-attachments/assets/e228b554-c1ce-4d10-aa12-9da9d8b7e6ad">

<img width="2054" alt="7  Add Sheets Data Source" src="https://github.com/user-attachments/assets/409e9a16-9dc0-4eb4-8c84-b32a0853b5f1">

<img width="2052" alt="8  Add Shopify Data Source" src="https://github.com/user-attachments/assets/092751c5-5d9d-42d3-92d8-de04f96fbcac">
